src/components/hub/EncryptedAuth.js

```diff
@@ -0,0 +1,35 @@
+import encryptionService from "@services/encryptionService";
+import showCustomAlert from "@utils/show_alert";
+import { useRef } from "react";
+
+import "@styles/hub/EncryptedHub.css";
+
+import { ReactComponent as EncryptedConversation } from "@icons/EncryptedConversation.svg";
+
+export default function EncryptedAuth() {
+  const inputRef = useRef(null);
+
+  const registerDeviceFunc = async () => {
```

Comment:
`registerDeviceFunc -> unlock`

Also, in ideal case, we could do an app lock each 5 mins w/o user inactivity.

Hence this function should **Not** create a new device each time the unlock happens.
It just needs to unpickl the account from storage.